src/main/java/redis/clients/util/JedisClusterHashTagUtil.java

```diff
@@ -0,0 +1,45 @@
+package redis.clients.util;
+
+/**
+ * Holds various methods/utilities to manipualte and parse redis hash-tags.
+ * See <a
+ * href="http://redis.io/topics/cluster-spec">Cluster-Spec : Keys hash tags</a>
+ */
+public final class JedisClusterHashTagUtil {
+
+  private JedisClusterHashTagUtil() {
+    throw new InstantiationError("Must not instantiate this class");
+  }
+
+  public static String getHashTag(String key) {
+    return extractHashTag(key, false, false);
+  }
+
+  public static boolean isClusterCompliantMatchPattern(String matchPattern) {
+    String tag = extractHashTag(matchPattern, true, true);
+    if (tag != null && !tag.isEmpty() && matchPattern.startsWith(tag)) {
+      return matchPattern.substring(tag.length()).equals("*");
```

Comment:
@HeartSaVioR  I got what you were trying to say now. 

I ran some more tests and noticed that tag doesn't have to be at the start/prefix of the pattern for it to work . In fact , as long as there is a tag inside the match pattern , I extract it and pass it to the Jedis SDK to extract the correct slot and the SCANing will work as usuall . Therefor - as long the match pattern contains a tag , it will be considered as valid 

``
127.0.0.1:5000> SET key1{FF}1yek 46
OK
127.0.0.1:5000> SET key2{FF}2yek 646
OK
127.0.0.1:5000> SET key3{FF}3yek 64364
OK
127.0.0.1:5000> SCAN 0 MATCH k*{FF}* COUNT 2000
1) "0"
2) 1) "key2{FF}2yek"
   2) "key1{FF}1yek"
   3) "key3{FF}3yek"
``